composables/drop/massmint/useDropMassMint.ts

```diff
@@ -0,0 +1,265 @@
+    } catch (error) {
+      console.log('[MASSMINT::GENERATE] Failed', error)
+      isLoading.value = false
+    }
+  }
+
+  const allocate = async (mintNfts: MassMintNFT[]) => {
+    try {
+      isLoading.value = true
+      isAllocating.value = true
+
+      const email = raffleEmail.value || getFakeEmail()
+      const address = accountId.value
+
+      const items = mintNfts.map(({ image, hash, metadata }) => ({
+        image,
+        hash,
+        metadata,
+      }))
+
+      allocatedNFTs.value = await allocateItems({ items, email, address })
+
+      // even thought the user might want x amount of items the worker can return a different amount
+      const allocatedNFTsToMint = toMintNFTs.value.slice(
+        0,
+        allocatedNFTs.value.length,
+      )
+
+      toMintNFTs.value = allocatedNFTsToMint.map((toMint, index) => {
+        const allocated = allocatedNFTs.value[index]
+        return allocated
+          ? { ...toMint, name: allocated.name, sn: Number(allocated.id) }
+          : toMint
+      })
+    } catch (error) {
+      console.log('[MASSMINT::ALLOCATE] Failed', error)
+    } finally {
+      isAllocating.value = false
+      isLoading.value = false
+    }
+  }
+
+  const allocateItems = async ({
+    items,
+    email,
+    address,
+  }): Promise<AllocatedNFT[]> => {
+    const results = [] as Array<AllocatedNFT>
+
+    // @see https://github.com/kodadot/private-workers/pull/86#issuecomment-1975842570 for context
+    for (const item of items) {
+      const { result } = await allocateCollection(
+        {
+          email,
+          address,
+          image: item.image,
+          hash: item.hash,
+          metadata: item.metadata,
+        },
+        drop.value.id,
+      )
+      results.push(result)
+    }
+
+    return results
+  }
+
+  const submitMint = async (nft: MassMintNFT): Promise<DoResult> => {
+    return new Promise((resolve, reject) => {
+      try {
+        claimAllocation(
+          {
+            sn: nft.sn,
+            txHash: mintingSession.value.txHash,
+            address: accountId.value,
+          },
+          drop.value.id,
+        ).then(({ result }) => resolve(result))
+      } catch (e) {
+        reject(e)
+      }
+    })
+  }
+
+  watch(allPinned, async (pinned) => {
+    if (pinned) {
+      await allocate(toMintNFTs.value)
+    }
+  })
+
+  watch([status, () => mintingSession.value.txHash], ([curStatus, txHash]) => {
+    if (curStatus === TransactionStatus.Cancelled) {
+      return onTransactionCancel?.()
+    }
+
+    // ensure txHash is set, it's needed when calling /do/:id
+    if (curStatus === TransactionStatus.Block && txHash) {
+      if (isError.value) {
+        isLoading.value = false
+        isTransactionLoading.value = false
+        return
+      }
+      onSubmitMints()
+    }
+  })
+
+  return {
+    raffleEmail,
+    canMint,
+    canListMintedNfts,
+    subscribeForNftsWithMetadata,
```

Comment:
do not return refs that come from another composable
composition can be done in the components


same as you don't return `accountId` from this composable